src/clover/fileparser.py

```diff
@@ -2176,6 +2176,79 @@ def _parse_transmission_inputs(
         transmitters,
     )
 
+# Grid inputs file:
+#   Keyword used for parsing diesel-generator information.
+GRID: str = "grid"
+GRID_EMISSIONS: str="emissions"
+GridTier: str="tier"
+GridType: str="type"
+# # The relative path to the grid inputs file.
+
+GRID_INPUTS_FILE: str = os.path.join("generation", "grid_inputs.yaml")
+
+def _parse_grid_inputs(
+    inputs_directory_relative_path: str,
+    logger: Logger,
+    scenarios: List[Scenario],
+) -> List[GRID]:
+
+    grid_inputs_filepath = os.path.join(
+        inputs_directory_relative_path, GRID_INPUTS_FILE
+    )
+    grid_inputs = read_yaml(grid_inputs_filepath, logger)
+    if not isinstance(grid_inputs, dict):
+        raise InputFileError(
+            "Grid inputs", "Grid input file is not of type `list`."
+        )
+    logger.info("Grid inputs successfully parsed.")
+
+    grids: List[Grid] = []
+    for entry in grid_inputs:
+        tiers: List[GridTier] = []
+        for tier_entry in entry["tiers"]:
+            tiers.append(
+                GridTier(
```

Comment:
And, and I'm not sure whether you've tested this, this will break here with something like `AttributeError` or `TypeError` because you can't call a string 😉

Reply:
Yes, this is all part of the same comment really about overriding the `GridTier` class :smile: